### ISSUE TITLE
False positive `unused-private-member` if class member is mutated via `cls`

### ISSUE THREAD

**Issue by gcolombeau** [PyCQA/pylint#4657]
### Steps to reproduce

Given a file `w0238.py`:
```python
"""unused-private-member (W0238)"""

class UnusedPrivateMember:
    """unused-private-member (W0238)"""

    __attr_a = None
    __attr_b = 'b'

    @classmethod
    def load_attrs(cls):
        """Load attributes."""
        cls.__attr_a = 'a'

    @property
    def attr_a(self):
        """Get a."""
        return self.__attr_a

    @property
    def attr_b(self):
        """Get b."""
        return self.__attr_b

```

### Current behavior

Result of `pylint w0238.py`:
```
************* Module w0238
w0238.py:12:8: W0238: Unused private member `UnusedPrivateMember.__attr_a` (unused-private-member)
```

### Expected behavior

I would expect no `unused-private-member` warning as `cls` is inferred as the current class. 
However, this example feels error prone under edge cases.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.3
astroid 2.6.2
Python 3.8.3 (default, Jul  3 2020, 09:39:00) 
[GCC 7.5.0]
```

**Comment by yushao2** [comment]
seems similar to #4644 , thanks for reporting this!

**Comment by Pierre-Sassoulas** [comment]
Do you want to handle this one @yushao2 ?

**Comment by alt-yushao2** [comment]
> Do you want to handle this one @yushao2 ?

Yep, will work on it later
